int16 actual_right
int16 actual_left
float32 delta_time